models/baseModels/Invoice/Invoice.ts

```diff
@@ -689,6 +757,34 @@ export abstract class Invoice extends Transactional {
     return transfers;
   }
 
+  async _validateHasLinkedReturnInvoices() {
+    const returnInvoices = await this._getLinkedReturnInvoiceNames();
+
+    if (!returnInvoices?.length) {
+      return;
+    }
+
+    const names = returnInvoices.map(({ name }) => name).join(', ');
+    const label =
+      this.fyo.schemaMap[this.schema.name]?.label ?? this.schema.name;
+    throw new ValidationError(
+      this.fyo.t`Cannot cancel ${this.schema.label} ${this
```

Comment:
`this.schema.label` and `label` would be the same value, also for the latter it should mention that those are return invoices.